internal/dangling/dangling_test.go

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dangling
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	nropv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1"
+	nodegroupv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1/helper/nodegroup"
+	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+const (
+	nrouid  = "123"
+	ns      = "nro-ns"
+	nroName = "nro"
+)
+
+var (
+	targetMCPName = "mcp1" // var because we need its pointer
```

Comment:
why can't it be local?

Reply:
we need it in the shared trees object which is also local:
https://github.com/openshift-kni/numaresources-operator/pull/1159/files/7dc9836f111e8f9b9affd75d66fb3e5aec705013#diff-62cf9676d59b182225fe67db37903c5d08981a360186170552ab10f6b0b7dd48R56

Reply:
this seems a great reason to have local variables instead of one shared global

Reply:
I get your point and it is valid. we are not updating this variable's value but we still need it as a variable because we want its pointer. I'll change that to be local var instead to avoid confusion or misuse if it happens.